src-openmp/datasets.cpp

```diff
@@ -527,11 +527,14 @@ DataFrame DataFrame::sample(int nrow, int seed, bool replace) const{
         // Draw row indices from uniform distribution
         std::uniform_int_distribution<> distr(0, this->length()-1);
         // pull random rows (as pointers, not copies) with replacement until full
-        while (new_frame.length() < nrow){
-            // get random row index with replacement
-            int rand_row = distr(eng);
-            // get pointer to that row in original dataframe and store in bootstrap
-            new_frame.addRow(this->row(rand_row));
+        #pragma omp parallel shared(new_frame)
```

Comment:
I believe `#pragma omp parallel` just creates a team of threads for parallel execution. We likely need another directive to actually start the parallel execution (similar to `#pragma omp for`). Do you know how to do that for a while loop?

Reply:
As far as I know, based on a StackOverflow post, simply wrapping the whole loop inside a pragma omp parallel should automatically parallelise the whole loop. As such, the omp pragma for would be redundant unless we want to specify shared or private objects. Not sure, though. The performance gain from that specific loop is also so marginal it was hard to tell from the little time left that it takes to run the script for 500 trees.